### DISCUSSION TITLE
How to plug an SDK into VitePress

### DISCUSSION THREAD

**Discussion by erondpowell** [vuejs/vitepress#1856] [Q&A]
I'd like to plug an SDK into Vitepress. 
I'm assuming I should just pass it as `app.use(mySDK)` in `enhanceApp` in index.js
Is there anything to be warned or noted when doing that for vitepress?

I'll use it to fetch and render some markdown in a db (shold happen serverside).
Later, it might have dynamic CRUD operations or a "login" element (clientside).
Are there any tips with getting the data to render serverside?

**Comment by brc-dd** [comment]
Client side stuff can directly be put in script blocks in markdown/vue files. If you need to do some server side stuff at build only, then you can use `.data.ts` files (ref: https://github.com/vuejs/vitepress/tree/main/__tests__/e2e/static-data, https://github.com/vuejs/vitepress/issues/794#issuecomment-1230986480, https://github.com/vuejs/vitepress/discussions/1291#discussioncomment-3989793). VitePress is a SSG, so it won't be possible to run something at server after build (because there is no server). If that's a requirement, you can use Nuxt with some docs-oriented theme like this: https://github.com/nuxt-themes/docus

**Reply by erondpowell** [reply]
@brc-dd  Thanks for the tips.

Yes, I meant to say "at build", not server side!
I wanna fetch data via SDK at build, then do everything on the client side after that.
